clippy_lints/src/single_component_path_imports.rs

```diff
@@ -39,6 +39,7 @@ declare_lint_pass!(SingleComponentPathImports => [SINGLE_COMPONENT_PATH_IMPORTS]
 impl EarlyLintPass for SingleComponentPathImports {
     fn check_item(&mut self, cx: &EarlyContext<'_>, item: &Item) {
         if_chain! {
+            if !in_macro(item.span);
```

Comment:
Shouldn't we also exclude external macros?

Reply:
`in_macro` should ignore any macros including external macros.

Reply:
Ah gotcha! I thought it was meant only for internal macros. My bad.